sklearn/model_selection/_validation.py

```diff
@@ -492,6 +493,43 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
     return ret
 
 
+def _score_single_metric(estimator, X_test, y_test, scorer):
+    """Copmute the score of an estimator given a test set and a single metric.
+    This function will also check whether the estimator has a classes attribute
+    and pass that into the labels attribute of the scorer if available.
+    """
+    # check if estimator has a classes_ attribute and scorer has labels attr:
+    est_has_classes = hasattr(estimator, "classes_")
+    if hasattr(scorer, "_score_func"):
+        scorer_has_labels = "labels" in signature(
+                                        scorer._score_func).parameters
+    else:
+        scorer_has_labels = False
+
+    if est_has_classes & scorer_has_labels:
+        labels = getattr(estimator, "classes_", None)
+        scorer._set_score_func_parameters(labels=labels)
```

Comment:
But currently an estimator's classes_ may be incomplete, and labels may already be set in a scorer, although not when getting a score by string. These cases need to be tested.

Also, I think we're modifying a public object here.